proposals/nnnn-streaming-media-manager-update.md

```diff
@@ -0,0 +1,148 @@
+FULL        | Foreground        | Yes                   | No
+FULL        | Resigning Active  | No                    | No
+
+###### Audio Stream Lifecycle Chart
+Since the app state is irrelevant for audio streaming (because we can do this in the background), the lifecycle chart is a bit simpler than the video lifecycle chart.
+
+HMI State   | Can Open Audio Stream | Should Close Audio Stream
+------------|-----------------------|--------------------------
+NONE        | No                    | Yes
+BACKGROUND  | No                    | Yes
+LIMITED     | Yes                   | No
+FULL        | Yes                   | No
+
+##### Sending Video Data
+The way in which video frames are sent over will not change from the current implementation. This function will require input of a `CVImageBufferRef` for video. It will be up to the developer to decide how to frequently to call this function. The developer will be able to know if the currently connected stream is encrypted with a `BOOL` `isEncrypted`, a `CGSize` `screenSize` to let them know the current Head Unit screen size, and a `CVPixelBufferPoolRef` for using the encoder’s pixel buffer pool, as is currently available in SMM.
+There will also be getter `isVideoStreamingPaused` that informs a developer to if sending of video data is not occurring based on circumstances where there’s not a need to send frames. If video streaming moves to a state where streaming cannot occur (i.e. HMI state changes to Limited, or Phone state move to Resigning Active), `isVideoStreamingPaused` will be set to `YES`, and will resume when possible.
+
+The thought for implementation is as follows:
+```objc
+// Checking if SMM is paused
+if (streamingMediaManager.isVideoStreamingPaused) {
+	<#pause pulling of frames if running#>
+	return;
+}
+
+// Sending Video Data
+CVPixelBufferPoolRef pixelBufferPool = streamingMediaManager.pixelBufferPool;
+CGSize screenSize = streamingMediaManager.screenSize;
+
+CVImageBufferRef frame = <#acquire video frame#>
+[streamingMediaManager sendVideoFrame:frame];
+```
+
+##### Sending video data when iPhone is moving to background
+Since iOS cannot use the encoder when the phone is in the background, to conserve battery and CPU, we must find a solution that will drive the user to reopen the app. The suggested solution is to create a single frame that is generated when SMM’s encoder is initialized, and send that over as the Phone State changes from **Active** to **Resigning Active**. This should be built as a non-customizable screen, so that we can rely on consistent behavior among streaming applications for the time being. The current idea is to have a black screen with the text “Please re-open <#App Name#>”. We have the option to provide translations for all languages that SDL supports, if we wish to go that route. We can also use the app icon, if we wish instead of the app name, as the app name could potentially be quite long.
```

Comment:
I would say both name and icon could be helpful. Do we have an idea for how this might be implemented?